### PR TITLE
updated ansible/ansible:vcenter-simulator Dockerfile to golang base

### PR DESCRIPTION
**Pull request by dericcrago**
also added exact versions of govmomi (specific hash) and python packages (requirements.txt)

this continues to use vcsim from commit dee49fa on Oct 13, 2017

### CODE REVIEW

**Comment by dericcrago** [comment]
Passed all ``"v{center,mware}_*"`` module integration tests when ran locally.

```bash
vcenter_folder
PLAY RECAP *********************************************************************
testhost                   : ok=17   changed=3    unreachable=0    failed=0   

vcenter_license
PLAY RECAP *********************************************************************
testhost                   : ok=9    changed=1    unreachable=0    failed=0   

vmware_cluster
PLAY RECAP *********************************************************************
testhost                   : ok=16   changed=2    unreachable=0    failed=0   

vmware_datacenter
PLAY RECAP *********************************************************************
testhost                   : ok=16   changed=2    unreachable=0    failed=0   

vmware_datastore_cluster
PLAY RECAP *********************************************************************
testhost                   : ok=19   changed=4    unreachable=0    failed=0   

vmware_datastore_facts
PLAY RECAP *********************************************************************
testhost                   : ok=23   changed=1    unreachable=0    failed=0   

vmware_datastore_maintenancemode
vmware_drs_rule_facts
PLAY RECAP *********************************************************************
testhost                   : ok=15   changed=1    unreachable=0    failed=0   

vmware_dvs_portgroup
PLAY RECAP *********************************************************************
testhost                   : ok=27   changed=7    unreachable=0    failed=0   

vmware_dvswitch
PLAY RECAP *********************************************************************
testhost                   : ok=20   changed=2    unreachable=0    failed=0   

vmware_guest
PLAY RECAP *********************************************************************
testhost                   : ok=251  changed=27   unreachable=0    failed=0   

vmware_guest_disk_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_guest_facts
PLAY RECAP *********************************************************************
testhost                   : ok=25   changed=1    unreachable=0    failed=0   

vmware_guest_find
PLAY RECAP *********************************************************************
testhost                   : ok=19   changed=1    unreachable=0    failed=0   

vmware_guest_powerstate
PLAY RECAP *********************************************************************
testhost                   : ok=26   changed=1    unreachable=0    failed=0   

vmware_guest_snapshot
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_guest_snapshot_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_guest_tools_wait
PLAY RECAP *********************************************************************
testhost                   : ok=11   changed=2    unreachable=0    failed=0   

vmware_host
PLAY RECAP *********************************************************************
testhost                   : ok=23   changed=3    unreachable=0    failed=0   

vmware_host_acceptance
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_capability_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_config_facts
PLAY RECAP *********************************************************************
testhost                   : ok=19   changed=1    unreachable=0    failed=0   

vmware_host_config_manager
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_dns_facts
PLAY RECAP *********************************************************************
testhost                   : ok=19   changed=1    unreachable=0    failed=0   

vmware_host_facts
PLAY RECAP *********************************************************************
testhost                   : ok=10   changed=1    unreachable=0    failed=0   

vmware_host_firewall_facts
PLAY RECAP *********************************************************************
testhost                   : ok=19   changed=1    unreachable=0    failed=0   

vmware_host_firewall_manager
PLAY RECAP *********************************************************************
testhost                   : ok=23   changed=3    unreachable=0    failed=0   

vmware_host_ntp
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_package_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_service_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_service_manager
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_host_vmnic_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_local_role_manager
PLAY RECAP *********************************************************************
testhost                   : ok=21   changed=5    unreachable=0    failed=0   

vmware_local_user_manager
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_maintenancemode
PLAY RECAP *********************************************************************
testhost                   : ok=21   changed=3    unreachable=0    failed=0   

vmware_resource_pool
PLAY RECAP *********************************************************************
testhost                   : ok=25   changed=3    unreachable=0    failed=0   

vmware_target_canonical_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_vm_facts
PLAY RECAP *********************************************************************
testhost                   : ok=13   changed=1    unreachable=0    failed=0   

vmware_vm_vm_drs_rule
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_vmkernel_facts
PLAY RECAP *********************************************************************
testhost                   : ok=1    changed=0    unreachable=0    failed=0   

vmware_vswitch
PLAY RECAP *********************************************************************
testhost                   : ok=18   changed=3    unreachable=0    failed=0   
```

**Comment by jctanner** [comment]
@mattclay shipit

**Comment by mattclay** [comment]
Before merging these improvements I'd like to merge https://github.com/ansible/vcenter-test-container/pull/3, which contains the minimal changes needed to pin requirements to match our currently deployed simulator.

**Comment by mattclay** [comment]
@dericcrago Ready for you to rebase this PR.

**Comment by dericcrago** [comment]
@mattclay - rebased and successfully reran tests locally

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,11 @@
+bzr==2.8.0.dev1
```
Comment (mattclay) [inline requirements.txt]: Some of these requirements appear to be unrelated to the Flask or psutil install. If you install both of those in a fresh venv you should get the current list of up-to-date requirements needed.

File: Dockerfile [hunk 2]
```diff
@@ -1,33 +1,11 @@
...
 
-RUN (cd /lib/systemd/system/sysinit.target.wants/; for i in *; do [ $i == systemd-tmpfiles-setup.service ] || rm -f $i; done); \
-rm -f /lib/systemd/system/multi-user.target.wants/*; \
-rm -f /etc/systemd/system/*.wants/*; \
-rm -f /lib/systemd/system/local-fs.target.wants/*; \
-rm -f /lib/systemd/system/sockets.target.wants/*udev*; \
-rm -f /lib/systemd/system/sockets.target.wants/*initctl*; \
-rm -f /lib/systemd/system/basic.target.wants/*; \
-rm -f /lib/systemd/system/anaconda.target.wants/*;
+ENV container=docker
```
Comment (mattclay) [inline Dockerfile]: This var shouldn't be needed. It was copied from our other test containers.

**Review by mattclay — CHANGES_REQUESTED** [review]

**Comment by dericcrago** [comment]
@mattclay - good catches, updated!

**Review by mattclay — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,33 +1,9 @@
-FROM fedora:26
+FROM golang:1
```
Comment (mattclay) [inline Dockerfile]: Actually, I didn't pay close enough attention to this before. This should be a specific version so we have repeatable builds. It looks like version 1 references the latest 1.x build.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (dericcrago) [reply Dockerfile]: How specific would you like? 1.10 or 1.10.1 ?

Comment (dericcrago) [reply Dockerfile]: Added ``golang:1.10.1``

Comment (mattclay) [reply Dockerfile]: I'm OK with more specific for now. My primary concern is having repeatable builds in the future. The more specific the versions are, the fewer issues we'll encounter when having to update and/or rebuild older versions of our test containers. It also helps with troubleshooting issues with tests, as it's very obvious what has changed between test container versions.

**Review by mattclay — APPROVED** [review]
